docs/proposals/probabilistic_proofs.md

```diff
@@ -0,0 +1,241 @@
+- **Failure** (the remainder of the sample space excluding Success):
+  - Servicer submits a true claim and is required prove it
+  - Servicer a true claim and have no requirement to prove it
+  - Servicer submits a false claim and gets away with it
+  - Servicer submits a true claim, but fails to prove it
+
+<!--
+The types of questions we can ask are:
+
+- What is the stopping condition?
+- How many success until a single failure?
+- How many failures until a single success?
+- How many failures until N success?
+- How many success until N failures?
+- How much scalability does the network need?
+- With what likelihood should an attacker get away?
+-->
+
+### Example
+
+Assume:
+
+- `ProofRequirementThreshold = 100 POKT`
+- `ProofRequestProbability = 0.25`
+
+If the `Claim` is greater than or equal to `100 POKT`, a proof is mandatory and the model is _"short-circuited"_. Therefore, the attacker can _freeload_ by submitting claims for `99.99 POKT` and hope they never get caught. If they do get caught (i.e. `Success`), the burn (i.e. `ProofMissingPenalty`) should exceed the total reward accumulated.
+
+Since each claim is independent, an attacker would never submit a `Claim` exceeding `ProofRequirementThreshold`, and therefore have a `ProofRequestProbability` likelihood of being required to submit a proof.
+
+### Model
+
+A [Geometric_distribution](https://en.wikipedia.org/wiki/Geometric_distribution) is used to identify the probability of `k` failures (sample space containing an attacker getting away) until a single success (an attacker is caught).
+
+$$ p = ProofRequestProbability $$
+
+$$ q = 1 - p $$
+
+$$ Pr(X=k) = (1-p)^{k-1}p $$
+
+$$ k = \frac{ln(\frac{Pr(X=k)}{p})}{ln(1-p)} + 1 $$
+
+![download](https://user-images.githubusercontent.com/1892194/220803154-90dcdd6b-8141-40d2-9cca-ed27a995fcfb.png)
+
+### Selecting Values
+
+#### Calculation
+
+_TODO(olshansky): Look at claims data to figure out the value for ProofRequirementThreshold_
+
+`ProofRequirementThreshold` should be as small as possible so that most such that most Claims for into the probabilistic bucket, while also balancing out penalties that may be too large for faulty honest Servicers. It will be selected to be `2σ` above the Claim `μ` such that `97.3%` fall into the `ProofRequestProbability` part of the piecewise function.
+
+`ProofRequestProbability (p)` is selected as `0.25` to enable scaling the network by `4x`.
+
+`Pr(X=k) > 0` must be greater than 0 while minimizing `k` as much as possible, so it is selected at approximately `0.01.`.
+
+$$ k = \frac{ln(\frac{Pr(X=k)}{p})}{ln(1-p)} + 1 $$
+
+$$ k = \frac{ln(\frac{0.01}{0.25})}{ln(1-0.25)} + 1 $$
+
+$$ k ≈ 12.19 $$
+
+Selecting `k = 12` implies that there is an `~1%` of 12 failures (includes the sample space where an attacker gets away) until a single success (an attacker is caught). To deter this behaviour, the `ProofMissingPenalty` should be selected as 12 times `ProofRequirementThreshold`.
```

Comment:
1% is the chance of doing exactly k=12 failures before a success, what you really want to know is the chance of any attacker getting 12 or more failures before a success. You should use the CDF to get this value, with p=025 and k=12 the CDF will tell you that 96.86% of the cases will fall here. If you want to get the 99% of the cases you should get it from the CDF:
0.99 <= 1-((1-p)**k)  ; p=0.25
-> k =  np.ceil( np.log(1.-0.99)/np.log(1.-p) ) = 17

Reply:
Thanks for the feedback @RawthiL!

100% agree on using a CDF instead of a PDF here.

The most difficult part for me is trying to figure out exactly what a success/failure is and be very clear about what question we're answering and why.

Rather than `any attacker getting 12 or more failures before a success`, what we'll be modelling with a CDF is `the protocol failing 12 times or less (i.e. not catching an attacker, or handling a normal claim) until a single success (i.e. catching an attacker)`

Does that make sense?

<img width="596" alt="Screenshot 2023-02-23 at 7 37 54 PM" src="https://user-images.githubusercontent.com/1892194/221086054-df25a888-558a-497e-9c13-87c6e07cbe6d.png">

Reply:
I think I'm comfortable with this formulation now

<img width="967" alt="Screenshot 2023-02-23 at 8 08 19 PM" src="https://user-images.githubusercontent.com/1892194/221089578-d37d8015-7af3-4edf-9c7a-e6b7789eb919.png">

Reply:
> the protocol failing 12 times or less (i.e. not catching an attacker, or handling a normal claim) until a single success (i.e. catching an attacker)

The CDF will tell us the chance that the protocol successfully catches an attacker in less than 12 consecutive failures. So, it makes sense but I would not word it as the protocol "failing" K times, it is just "trusting". So maybe it could be worded like this?:
> the protocol successfully catching an attacker (or challenging a normal claim) before 12 (or more) consecutive unchallenged claims (i.e. failing catching an attacker or correctly trusting the node).

The graph is OK now.

In the pictor I see that you updated the K to use the CDF, this is OK but you still include the paragraph that models the claims as normal distributions (thats not the case as we observed).

Reply:
I made the changes below based on your feedback. Let me know if its clearer now

---

> **The question being answered by the distribution**: What is the probability of the protocol trusting (i.e. failing) `k` Claims or less (i.e. handling a normal claim or not catching an attacker) until a single penalty enforcement (i.e. successfully catching an attacker).
> 
> **Answer**: Selecting `k = 16` and `ProofRequirementThreshold = 20 POKT` implies that if an attacker continues submitting claims for `19.99 POKT` or less, they will get caught `99%` of the time, and will be penalized for `320 POKT`.

---


> `ProofRequirementThreshold` should be as small as possible so that most such that most Claims for into the probabilistic bucket, while also balancing out penalties that may be too large for faulty honest Servicers. Ideally, it should be selected to be `2σ` above the Claim `μ` such that `97.3%` fall into the `ProofRequestProbability` part of the piecewise function. However, as seen in the Appendix, the POKT Claim distribution does not follow a normal distribution. Instead, 20 POKT was selected since it is greater than `p95` of POKT claim using the data collected.

Reply:
Great!